service/ec2/customizations.go

```diff
@@ -8,7 +8,43 @@ import (
 	"github.com/aws/aws-sdk-go-v2/internal/awsutil"
 )
 
+type retryer struct {
+	aws.DefaultRetryer
+}
+
+func (d retryer) RetryRules(r *request.Request) time.Duration {
+	switch r.Operation.Name {
+	case opModifyNetworkInterfaceAttribute:
+		fallthrough
+	case opAssignPrivateIpAddresses:
+		return d.customRetryRule(r)
+	default:
+		return d.DefaultRetryer.RetryRules(r)
```

Comment:
We could remove the need of having two code paths usages of the same utility by only adding the custom retrier for the API operations that it is needed for. This would mean the custom retrier would be added in `initRequest` instead of `initClient`. And would use this switch above to determine when to add the custom retrier to a request.